packages/shell-api/src/integration.spec.ts

```diff
@@ -983,6 +983,31 @@ describe('Shell API (integration)', function() {
           }
         ]);
       });
+      it('creates a view that potentially contains JS functions in its pipeline', async() => {
+        const pipeline = (body: any) => [{
+          '$set': {
```

Comment:
I see that tests are failing, maybe this should be `addFields` instead?

```suggestion
          '$addFields': {
```

Reply:
Oh, I’m also fairly clueless here. This passed locally and was taken directly from the ticket, and it looks like it’s the cli-service-provider tests that are failing (because I didn’t run *them* locally, which is entirely my fault, but … :shrug:)

Reply:
Did some googling around and seems like `$set` is a [new feature in 4.2](https://docs.mongodb.com/manual/reference/operator/aggregation/set/) and `$function` is [new in 4.4](https://docs.mongodb.com/manual/reference/operator/aggregation/function/). Should we skip that test for anything that is less than that?

Reply:
just pushed a commit for that, yes :)

Reply:
Haha, ~ at the same time, perfect timing 😄